compiler/luci-interpreter/src/kernels/If.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/If.h"
+
+#include <cstring>
+
+namespace luci_interpreter
+{
+namespace kernels
+{
+
+If::If(const Tensor *cond, std::vector<const Tensor *> inputs, std::vector<Tensor *> outputs,
+       RuntimeGraph *then_graph, RuntimeGraph *else_graph)
+    : _cond(cond), _inputs(std::move(inputs)), _outputs(std::move(outputs)),
+      _then_graph(then_graph), _else_graph(else_graph)
+{
+}
+
+void If::configure()
+{
+  assert(_cond->element_type() == DataType::BOOL);
+  assert(_cond->shape().num_elements() == 1);
+
+  // Configure both graphs.
+  for (RuntimeGraph *graph : {_then_graph, _else_graph})
+  {
+    assert(graph->getInputTensors().size() == _inputs.size());
+    assert(graph->getOutputTensors().size() == _outputs.size());
+
+    // Resize graph input tensors.
+    const auto &input_tensors = graph->getInputTensors();
+    for (size_t i = 0; i < _inputs.size(); ++i)
+    {
+      assert(input_tensors[i]->element_type() == _inputs[i]->element_type());
+      input_tensors[i]->resize(_inputs[i]->shape());
+    }
+    graph->configure();
+  }
+
+  // For now, output tensors of both graphs should have the same shape.
```

Comment:
Maybe they should have the same type as well?

Reply:
I added an assertion.